pkg/ccl/backupccl/key_rewriter.go

```diff
@@ -273,7 +274,19 @@ func (kr *KeyRewriter) RewriteKey(
 		copy(keyTenantPrefix, newTenantPrefix)
 		rekeyed = append(keyTenantPrefix, rekeyed...)
 	} else {
-		rekeyed = append(newTenantPrefix, rekeyed...)
+		l := len(newTenantPrefix) + len(rekeyed)
```

Comment:
could you write a comment explaining why you're doing this? I believe the idea is that you want to reduce the number of times we allocate memory by allocating a slab of memory every time we exhaust our alloc buf.

Reply:
it also seems that we can use this strategy here[ too](https://github.com/cockroachdb/cockroach/blob/master/pkg/ccl/backupccl/key_rewriter.go#L69), though I think this path in traditional restore.

Reply:
> that we can use this strategy here too

We could. we'd need to hang the slab off the rule instead of the rewriter or something. I'm not very motivated until I see it in a profile, which would probably be from backing up table < 127 and restoring > 127.

Reply:
added comments